packages/shorebird_cli/lib/src/commands/patch_command.dart

```diff
@@ -310,8 +324,42 @@ https://github.com/shorebirdtech/shorebird/issues/472
         return ExitCode.software.code;
       }
     }
+
+    final String releaseAabPath;
+    try {
+      releaseAabPath = await _downloadReleaseArtifact(
+        Uri.parse(releaseAabArtifact.url),
+      );
+    } catch (error) {
+      downloadReleaseArtifactProgress.fail('$error');
+      return ExitCode.software.code;
+    }
+
     downloadReleaseArtifactProgress.complete();
 
+    final contentDiffs = _aabDiffer.aabContentDifferences(
+      releaseAabPath,
+      bundlePath,
+    );
+
+    logger.detail('content diffs detected: $contentDiffs');
+
+    if (contentDiffs.contains(AabDifferences.native)) {
+      logger.err(
+        '''Your aab contains native changes, which cannot be applied in a patch. Please create a new release or revert these changes.''',
```

Comment:
Can we list the native changes? I'm thinking developers would likely want to know which changes to revert.

Reply:
We would need to be able to parse `.dex` files for this, which seems like a heavy lift https://source.android.com/docs/core/runtime/dex-format

Reply:
Ah yeah makes sense. Created https://github.com/shorebirdtech/shorebird/issues/539 and maybe we can revisit in the future?

Reply:
Similarly, I might say "The Android App Bundle appears to contain Kotlin or Java changes which cannot be applied via a patch.  Please create a new release or revert those changes to create a patch.  If you believe you're seeing this in error, please reach out to us for support.  shorebird.dev/support". (That url doesn't exist yet, but we should make something similar.)

Reply:
Done, and filed https://github.com/shorebirdtech/website/issues/15 to add a support page.